### PR TITLE
Add upgrade task

### PR DESCRIPTION
**Pull request by Daverlo**
I've extended the provided upgrade tasks by creating an abstract class with all the plumbing needed for it to work, and a couple of functions that help detect when we are dealing with a fresh install, as some upgrades might not be needed in that case.

It also add the first migration, that deals with the removed priority option and sets it in the workflow.

### CODE REVIEW

File: src/main/java/com/semmle/jira/addon/config/init/WorkflowUtils.java [hunk 1]
```diff
@@ -167,6 +194,26 @@ static void addLayoutToWorkflow(String workflowName, String layoutJson) {
     ofBizDelegator.createValue("OSPropertyText", textFields);
   }
 
+  private static void addFunction(ActionDescriptor transition, FunctionDescriptor function) {
+    @SuppressWarnings("unchecked")
+    List<FunctionDescriptor> functions = transition.getUnconditionalResult().getPostFunctions();
+    // remove existing functions with the same name
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/config/init/WorkflowUtils.java]: You don't set a name on the function  so the entire `remove existing functions with the same name` block could be removed.

Comment (Daverlo) [reply src/main/java/com/semmle/jira/addon/config/init/WorkflowUtils.java]: I'm setting the name in line 89, at the beginning of `addDefaultPriorityToWorkflow`.

File: src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask1CreateIssueType.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import com.semmle.jira.addon.util.JiraUtils;
+import java.util.Collection;
+import java.util.Collections;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@ExportAsService(PluginUpgradeTask.class)
+@Component
+public class UpgradeTask1CreateIssueType implements PluginUpgradeTask {
+  public final int BUILD_NUMER = 1;
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask1CreateIssueType.java]: I think the `BUILD_NUMBER`s could be declared `private`

File: src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask1CreateIssueType.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  @Override
+  public int getBuildNumber() {
+    return BUILD_NUMER;
+  }
+
+  @Override
+  public String getShortDescription() {
+    return "Upgrade Task " + BUILD_NUMER + " create 'LGTM aler' issue type.";
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask1CreateIssueType.java]: `aler` -> `alert`

File: src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask1CreateIssueType.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Collections;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@ExportAsService(PluginUpgradeTask.class)
+@Component
+public class UpgradeTask1CreateIssueType implements PluginUpgradeTask {
+  public final int BUILD_NUMER = 1;
+
+  protected final PluginSettings settings;
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask1CreateIssueType.java]: This field is not used, right? Please remove.

File: src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask2CreateWorkflow.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@ExportAsService(PluginUpgradeTask.class)
+@Component
+public class UpgradeTask2CreateWorkflow implements PluginUpgradeTask {
+  public final int BUILD_NUMER = 2;
+
+  protected final PluginSettings settings;
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask2CreateWorkflow.java]: This field is not used, please remove.

File: src/main/java/com/semmle/jira/addon/config/init/WorkflowUtils.java (hunk 1 above)
Comment (aibaars) [reply src/main/java/com/semmle/jira/addon/config/init/WorkflowUtils.java]: Ah yes indeed.

**Comment by aibaars** [comment]
Good work! Merging

**Pull request merged**